odoo/models.py

```diff
@@ -6168,9 +6169,17 @@ def filtered(self, func):
             records.filtered("partner_id.is_company")
         """
         if isinstance(func, str):
-            name = func
-            func = lambda rec: any(rec.mapped(name))
-        return self.browse([rec.id for rec in self if func(rec)])
+            path = func.split(".")
+
+            def get_value(rec):
+                for p in path:
+                    rec = rec[p]
+                return rec
```

Comment:
This diff changes the behavior.
e.g. the left `res_users.filtered('groups_id.name')` will return res_users who have at least one named group, but the modified one may raise error ('Expected singleton')
I think it might be a side-effect of the left method.
If we want to change that, maybe mention it in the commit message?